src/modules/flow/flow.c

```diff
@@ -176,18 +176,22 @@ static inline uint32_t compute_hessian_4x6(uint8_t *image, uint16_t x, uint16_t
  * @brief Compute the average pixel gradient of all horizontal and vertical steps
  *
  * TODO compute_diff is not appropriate for low-light mode images
+ * TODO calculate for 8x8 pattern instead of 4x4 to raise number of patches used for flow computation
```

Comment:
using a bigger pattern does not necessarily lead to more patches being used.

There is another issue with this function, it should compute the average pixel gradient, but it just sums up the gradient values. `acc` should be divided by the number of gradients computed.

Reply:
I think you are wrong here, right now one takes 8x8 patch and looks at only the center of the size 4x4 to determine if the patch should be skipped. That is only 1/4 area of the patch used, so if the patch has some pattern one can detect somewhere on the edge - this patch would still be skipped.
But you are right, doing such a change would require to adjust some thresholds.

Reply:
I'm not saying it does not do 4x4, but something else:
if the gradients are summed up as currently (wrongly) done, the output value trivially increases with a bigger patch, but it should use the average gradient to decide, which should in general make the output (and thus also the threshold) independent of the patch size.
On a conceptual level I agree that a bigger patch would allow to detect potential visual features at the borders of the 8x8 patch which would not be detected with 4x4.

I do not remember what the actual reason was to only do it with 4x4 - most likely it was lazyness "because 4x4 does a good enough job" ;).